StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/Button+Extensions.swift

```diff
@@ -63,3 +64,27 @@ extension Button.Configuration {
         return secondaryButtonConfiguration
     }
 }
+
+// attaches haptic feedback to a button press
+private final class ButtonFeedbackGeneratorHandler: NSObject {
+
+    @objc private func didTouchUpInside() {
+        FeedbackGeneratorAdapter.buttonTapped()
+    }
+
+    private static var associatedObjectKey: UInt8 = 0
+    static func attach(toButton button: UIControl) {
+        let buttonFeedbackGeneratorHandler = ButtonFeedbackGeneratorHandler()
+        objc_setAssociatedObject(
+            button,
+            &associatedObjectKey,
+            buttonFeedbackGeneratorHandler,
+            .OBJC_ASSOCIATION_RETAIN_NONATOMIC
+        )
+        button.addTarget(
+            buttonFeedbackGeneratorHandler,
+            action: #selector(didTouchUpInside),
+            for: .touchUpInside
+        )
+    }
```

Comment:
also if I'm wrong on anything happy to fix forward haha

Reply:
> in this case we are attaching a unique instance to each unique Button

> there are no "collisions" because they are separate instances

doesn't the fact that its a static var make it so there's only ever one for all instances of `Button`? 🤔 or am I missing something basic here?

Reply:
@bric-stripe 

I could be misunderstanding the question; another option is that you could do your breakdown of understanding of what's happening and then I could see where I'm off

> the fact that its a static var make it so there's only ever one for all instances 

there's only ever one address - so its correct to say that the "key" is uniquely the same for all instances

one can imagine `&associatedObjectKey` as being the same thing as "associated_object_key" String

and one can imagine each `Button` instance to be like a dictionary (where we can map keys to values)

`objc_setAssociatedObject(` gets passed in the _instance_ of the Button, so each instance of Button gets mapped a "key" (&associatedObjectKey) to a new instance of `ButtonFeedbackGeneratorHandler()`

---

I posted the code here:

```
class Button: UIControl {
	let associatedObjectKey = ButtonFeedbackGeneratorHandler()

	init() {
	   // wrong syntax psuedocode
	    addTarget(associatedObjectKey, ButtonFeedbackGeneratorHandler.didTouchUpInside)
	}
}
```

> static var make it so there's only ever one for all instances of 

this, if I read correctly, would imply:

```
class Button: UIControl {
	static let associatedObjectKey = ButtonFeedbackGeneratorHandler()
        var associatedObjectKey: ButtonFeedbackGeneratorHandler {
           return Self.associatedObjectKey
       }

	init() {
	   // wrong syntax psuedocode
	    addTarget(associatedObjectKey, ButtonFeedbackGeneratorHandler.didTouchUpInside)
	}
}
```

If I understand the question, I don't think that's true, but I could be wrong (!)